Extension/src/LanguageServer/client.ts

```diff
@@ -868,6 +877,94 @@ export class DefaultClient implements Client {
         return workspaceFolder ? workspaceFolder.name : "untitled";
     }
 
+    public updateClientConfigurations(): void {
+        clients.forEach(client => {
+            if (client instanceof DefaultClient) {
+                const defaultClient: DefaultClient = <DefaultClient>client;
+                defaultClient.configuration.CompilerDefaults = compilerDefaults;
+                defaultClient.configuration.handleConfigurationChange();
+            }
+        });
+    }
+
+    public set defaultCompiler(value: any) {
+        vscode.workspace.getConfiguration("files").update("associations", value, vscode.ConfigurationTarget.Global);
+    }
+
+    public async handleCompilerQuickPick(): Promise<void> {
+        const compilerPath: PersistentState<string[]> = new PersistentState<string[]>("CPP.trustedCompilerPaths", []);
+        const paths: string[] = compilerDefaults.knownCompilers.map(function (a: configs.KnownCompiler): string { return a.path; });
+        paths.push("Select another compiler on my machine");
+        paths.push("Help me install a compiler");
+        paths.push("Do not configure a compiler (not recommended)");
+        const index: number = await ui.showSelectCompiler(paths);
+        if (index === paths.length - 1) {
+            paths[index] = "";
+        } else if (index === paths.length - 2) {
+            switch (os.platform()) {
+                case 'win32':
+                    vscode.commands.executeCommand('vscode.open', "https://go.microsoft.com/fwlink/?linkid=2217614");
+                    return;
+                case 'darwin':
+                    vscode.commands.executeCommand('vscode.open', "https://go.microsoft.com/fwlink/?linkid=2217706");
+                    return;
+                case 'linux':
+                    vscode.commands.executeCommand('vscode.open', "https://go.microsoft.com/fwlink/?linkid=2217615");
+                    return;
+                default:
+                    throw new Error(localize("unexpected.os", "Unexpected OS"));
+            }
+        } else if (index === paths.length - 3) {
+            const result: vscode.Uri[] | undefined = await vscode.window.showOpenDialog();
+            if (result !== undefined && result.length > 0) {
+                util.addTrustedCompiler(result[0].fsPath);
+                await this.defaultCompiler(result[0].fsPath);
+                const selectedCompilerDefaults: configs.CompilerDefaults = await this.requestCompiler(compilerPath.Value);
+                compilerDefaults = selectedCompilerDefaults;
+                this.updateClientConfigurations();
+                return;
+            }
+        } else {
+            util.addTrustedCompiler(paths[index]);
+        }
+        await vscode.workspace.getConfiguration("C_Cpp.default").update("compilerPath", paths[index], vscode.ConfigurationTarget.Global);
```

Comment:
Shouldn't this go through the setter in settings.ts?

Reply:
Also, if "Do not configure a compiler (not recommended)" is selected, does this result in a value of `""` being set in settings.json?  If so, that may not be desirable, as that has a special meaning (that compilerPath should be explicitly disabled).

Reply:
What should it be set to instead?

Reply:
If the intention is to prompt them again next time, then the default.compilerPath should not get set to anything.  If the intention is to never prompt them again and disable the use of a compiler in the fallback default configuration, then `""` is appropriate.  If the intention is for some other behavior, we may need to identify what that behavior is and figure out how to get it.  What should the behavior be?  :)